### PR TITLE
Create switch case instead of if/else && Added import protection.

### PR DESCRIPTION
**Pull request by justnat3**
Bring this down and test. Please review this thoroughly.

### CODE REVIEW

**Comment by ghost** [comment]
This seems like extreme refactoring. I think if anything we just need more features/commands. Like an auto clear command. We have already refactored once and the last time we did it there were multiple errors that we didn't catch.

**Review by ghost — COMMENTED** [review]
Just something about this seems off. Like why name it switchcase. I cant read the code with githubs git diff but if i put it in a file im sure id find something

**Comment by justnat3** [comment]
> This seems like extreme refactoring. I think if anything we just need more features/commands. Like an auto clear command. We have already refactored once and the last time we did it there were multiple errors that we didn't catch.

a switch case statement would make scaling this a lot easier. I don't see how "excessive refactoring" as a good reason why it should not be pulled down and tested.

**Review by ghost — CHANGES_REQUESTED** [review]
in the discord py docs this is not the practice used as it is not needed and just added lines that do nothing.

**Comment by ghost** [comment]
> > This seems like extreme refactoring. I think if anything we just need more features/commands. Like an auto clear command. We have already refactored once and the last time we did it there were multiple errors that we didn't catch.
> 
> a switch case statement would make scaling this a lot easier. I don't see how "excessive refactoring" as a good reason why it should not be pulled down and tested.

No im not saying we shouldn't test it im just saying we dont need another refactor we just need more features

**Comment by justnat3** [comment]
> in the discord py docs this is not the practice used as it is not needed and just added lines that do nothing.

That is an example, there is no specific way to write this. Instead of checking each if statement it will choose the case based on what is feed into the switchcase function. How is this not more effective?

File: bot.py [hunk 1]
```diff
@@ -35,5 +35,7 @@ async def on_message(msg):
 
   }
   return switchcase.get(cmd, None)
-
-client.run(token)
+if __name__ == "__main__":
```
Comment (ghost) [inline bot.py:38]: This is not needed as it is only one line that we need to run. this is just extra lines that we dont need.

Comment (justnat3) [reply bot.py:38]: That's just fine. It is a practice. Not something that is going to change functionality

**Comment by ghost** [comment]
> > in the discord py docs this is not the practice used as it is not needed and just added lines that do nothing.
> 
> That is an example, there is no specific way to write this. Instead of checking each if statement it will choose the case based on what is feed into the switchcase function. How is this not more effective?

The code is already fast on its own. We just need more features.

**Comment by justnat3** [comment]
> > > in the discord py docs this is not the practice used as it is not needed and just added lines that do nothing.
> > 
> > 
> > That is an example, there is no specific way to write this. Instead of checking each if statement it will choose the case based on what is feed into the switchcase function. How is this not more effective?
> 
> The code is already fast on its own. We just need more features.

Scaling & Effectiveness. Not that it isn't fast already, the if statements are unnecessary.

**Comment by justnat3** [comment]
The ego is not something you should play around this. 


11:45 on 7/15/2020 - Chat log 

DoctorVem: My PR was based on pre-existing code. I'm not going to fight you about PRs, however, the title for your PR was very disrespectful.

insignia_microwave: @DoctorVem lol that was the point

**Pull request closed without merging**